cachi2/core/package_managers/pip.py

```diff
@@ -1891,18 +1891,9 @@ def _is_valid(pkg: pypi_simple.DistributionPackage) -> bool:
 
     if sdists:
         best_sdist = max(sdists, key=_sdist_preference)
+        processed_dpis.append(best_sdist)
         if best_sdist.is_yanked:
-            raise PackageRejected(
-                f"All sdists for package {name}=={version} are yanked",
-                solution=(
-                    f"Please update the {name} version in your requirements file.\n"
-                    "Usually, when a version gets yanked from PyPI, there will already "
-                    "be a fixed version available.\n"
-                    "Otherwise, you may need to pin to the previous version."
-                ),
-            )
-        if best_sdist:
-            processed_dpis.append(best_sdist)
+            log.warning("Please use a non-yanked version of the package")
```

Comment:
Nice! Very succinct.

Reply:
(Sorry, late to the party :sweat_smile:)

Just a minor observation, I don't this is a blocker for merging: we should probably add the name of the package to this line here, since the user can be filtering by WARNING, in which case he'd wouldn't understand what package we're talking about.

Here's how the logs look like in Cachi2:
 ```
2024-04-08 09:37:36,280 INFO -- Processing requirement line 'terminado==0.13.0'
2024-04-08 09:37:36,396 WARNING Please use a non-yanked version of the package
2024-04-08 09:37:36,482 INFO Verifying checksums of terminado-0.13.0.tar.gz
2024-04-08 09:37:36,483 INFO -- Finished processing requirement line 'terminado==0.13.0'
```

Reply:
Nice
```
2024-04-09 15:05:47,469 INFO -- Processing requirement line 'terminado==0.13.0'
2024-04-09 15:05:47,550 WARNING The version 0.13.0 of package terminado is yanked, use a different version
2024-04-09 15:05:47,648 INFO Verifying checksums of terminado-0.13.0.tar.gz
2024-04-09 15:05:47,650 INFO Verifying checksums of terminado-0.13.0-py3-none-any.whl
2024-04-09 15:05:47,651 INFO -- Finished processing requirement line 'terminado==0.13.0'
```